test/widgets/theme_test.dart

```diff
@@ -24,23 +23,19 @@ void main() {
       double? ambientTextScaleFactor,
     }) async {
       testWidgets(description, (WidgetTester tester) async {
+        addTearDown(testBinding.reset);
         if (ambientTextScaleFactor != null) {
           tester.platformDispatcher.textScaleFactorTestValue = ambientTextScaleFactor;
           addTearDown(tester.platformDispatcher.clearTextScaleFactorTestValue);
         }
-        late final double expectedFontSize;
-        late final double expectedLetterSpacing;
-        await tester.pumpWidget(
-          Builder(builder: (context) => MaterialApp(
-            theme: zulipThemeData(context),
-            home: Builder(builder: (context) {
-              expectedFontSize = Theme.of(context).textTheme.labelLarge!.fontSize!;
-              expectedLetterSpacing = proportionalLetterSpacing(context,
-                0.01, baseFontSize: expectedFontSize);
-              return button;
-            }))));
-
-        final text = tester.renderObject<RenderParagraph>(find.text(buttonText)).text;
+        await tester.pumpWidget(TestZulipApp(child: button));
+        await tester.pump();
+        final buttonTextFinder = find.text(buttonText);
+        final context = tester.element(buttonTextFinder);
+        final text = tester.renderObject<RenderParagraph>(buttonTextFinder).text;
```

Comment:
small simplification:
```suggestion
        final text = (context.renderObject as RenderParagraph).text;
```

Then also `buttonTextFinder` is used just once and can be inlined.

Reply:
also nit: reorder this line to just before the `check`, after computing the expected values; or perhaps even just inline it into the `check` call